cmd/vpp-agent-ctl/impl/etcd.go

```diff
@@ -0,0 +1,174 @@
+// be used to demonstrate the usage of VPP Agent plugins and their APIs.
+
+package impl
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+
+	"github.com/ligato/cn-infra/config"
+	"github.com/ligato/cn-infra/datasync"
+	"github.com/ligato/cn-infra/db/keyval"
+	"github.com/ligato/cn-infra/db/keyval/etcdv3"
+	"github.com/ligato/cn-infra/db/keyval/kvproto"
+)
+
+// CreateEtcdClient uses environment variable or ETCD config file to establish connection
+func (ctl *VppAgentCtl) CreateEtcdClient(configFile string) (*etcdv3.BytesConnectionEtcd, keyval.ProtoBroker, error) {
+	var err error
+
+	if configFile == "" {
+		configFile = os.Getenv("ETCDV3_CONFIG")
+	}
+
+	cfg := &etcdv3.Config{}
+	if configFile != "" {
+		err := config.ParseConfigFromYamlFile(configFile, cfg)
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	etcdConfig, err := etcdv3.ConfigToClientv3(cfg)
+	if err != nil {
+		ctl.Log.Fatal(err)
+	}
+
+	bDB, err := etcdv3.NewEtcdConnectionWithBytes(*etcdConfig, ctl.Log)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return bDB, kvproto.NewProtoWrapperWithSerializer(bDB, &keyval.SerializerJSON{}).
+		NewBroker(ctl.serviceLabel.GetAgentPrefix()), nil
+}
+
+// ListAllAgentKeys prints all keys stored in the broker
+func (ctl *VppAgentCtl) ListAllAgentKeys() {
+	ctl.Log.Debug("listAllAgentKeys")
+
+	it, err := ctl.broker.ListKeys(ctl.serviceLabel.GetAllAgentsPrefix())
+	if err != nil {
+		ctl.Log.Error(err)
+	}
+	for {
+		key, _, stop := it.GetNext()
+		if stop {
+			break
+		}
+		ctl.Log.Println("key: ", key)
+	}
+}
+
+// EtcdGet uses ETCD connection to get value for specific key
+func (ctl *VppAgentCtl) EtcdGet(key string) {
+	ctl.Log.Debug("GET ", key)
+
+	data, found, _, err := ctl.bytesConnection.GetValue(key)
+	if err != nil {
+		ctl.Log.Error(err)
+		return
+	}
+	if !found {
+		ctl.Log.Debug("No value found for the key", key)
+	}
+	ctl.Log.Println(string(data))
+}
+
+// EtcdPut stores key/data value
+func (ctl *VppAgentCtl) EtcdPut(key string, file string) {
+	input, err := ctl.readData(file)
+
+	ctl.Log.Println("DB putting ", key, " ", string(input))
+
+	err = ctl.bytesConnection.Put(key, input)
+	if err != nil {
+		ctl.Log.Panic("error putting the data ", key, " that to DB from ", file, ", err: ", err)
+	}
+	ctl.Log.Println("DB put successful ", key, " ", file)
+}
+
+// EtcdDel removes data under provided key
+func (ctl *VppAgentCtl) EtcdDel(key string) {
+	found, err := ctl.bytesConnection.Delete(key, datasync.WithPrefix())
+	if err != nil {
+		ctl.Log.Error(err)
+		return
+	}
+	if found {
+		ctl.Log.Debug("Data deleted:", key)
+	} else {
+		ctl.Log.Debug("No value found for the key", key)
+	}
+}
+
+// EtcdDump lists values under key. If no key is provided, all data is read.
+func (ctl *VppAgentCtl) EtcdDump(key string) {
+	ctl.Log.Debug("DUMP ", key)
+
+	data, err := ctl.bytesConnection.ListValues(key)
+	if err != nil {
+		ctl.Log.Error(err)
+		return
+	}
+
+	var found bool
+	for {
+		found = true
```

Comment:
Variable `found` doesn't make sense here since it's immediately set to true in the first loop.

Reply:
that's the original code, but I fix it